fvm/handler/programs.go

```diff
@@ -110,19 +110,19 @@ func (h *ProgramsHandler) Get(location common.Location) (*interpreter.Program, b
 }
 
 func (h *ProgramsHandler) Cleanup() error {
-
 	stackLen := len(h.viewsStack)
 
 	if stackLen == 0 {
 		return nil
 	}
 
-	for i := stackLen; i > 0; i-- {
+	for i := stackLen - 1; i > 0; i-- {
 		entry := h.viewsStack[i]
 		err := h.viewsStack[i-1].state.MergeState(entry.state)
 		if err != nil {
 			return fmt.Errorf("cannot merge state while cleanup: %w", err)
 		}
 	}
-	return h.masterState.State().MergeState(h.viewsStack[0].state)
+
+	return h.initialState.MergeState(h.viewsStack[0].state)
```

Comment:
So we always revert back to initial state?

Reply:
Yeah, we merge back all extra children that have been created by the program hander to capture touches.